k8s/http.go

```diff
@@ -0,0 +1,21 @@
+package k8s
+
+import "net/http"
+
+type dnsCheck interface {
+	Reachable(url string) error
+}
+
+type DefaultDNSChecker struct{}
```

Comment:
We don't have to export this

Reply:
Agreed, although this results in this linter comment from the `revive` linter:
```
 unexported-return: exported func NewDefaultDNSChecker returns unexported type *k8s.defaultDNSChecker, which can be annoying to use (revive)
```